target/ppc/translate/vmx-impl.c.inc

```diff
@@ -1008,41 +1008,263 @@ static void glue(gen_, name0##_##name1)(DisasContext *ctx)             \
+
+    tcg_gen_gvec_cmp(cond, vece, avr_full_offset(a->vrt),
+                     avr_full_offset(a->vra), avr_full_offset(a->vrb), 16, 16);
+    tcg_gen_gvec_shli(vece, avr_full_offset(a->vrt), avr_full_offset(a->vrt),
+                      (8 << vece) - 1, 16, 16);
+    tcg_gen_gvec_sari(vece, avr_full_offset(a->vrt), avr_full_offset(a->vrt),
+                      (8 << vece) - 1, 16, 16);
+
+    if (a->rc) {
+        do_vcmp_rc(a->vrt);
+    }
+
+    return true;
+}
+
+TRANS(VCMPEQUB, do_vcmp, TCG_COND_EQ, MO_8)
+TRANS(VCMPEQUH, do_vcmp, TCG_COND_EQ, MO_16)
+TRANS(VCMPEQUW, do_vcmp, TCG_COND_EQ, MO_32)
+TRANS(VCMPEQUD, do_vcmp, TCG_COND_EQ, MO_64)
+
+TRANS(VCMPGTSB, do_vcmp, TCG_COND_GT, MO_8)
+TRANS(VCMPGTSH, do_vcmp, TCG_COND_GT, MO_16)
+TRANS(VCMPGTSW, do_vcmp, TCG_COND_GT, MO_32)
+TRANS(VCMPGTSD, do_vcmp, TCG_COND_GT, MO_64)
+TRANS(VCMPGTUB, do_vcmp, TCG_COND_GTU, MO_8)
+TRANS(VCMPGTUH, do_vcmp, TCG_COND_GTU, MO_16)
+TRANS(VCMPGTUW, do_vcmp, TCG_COND_GTU, MO_32)
+TRANS(VCMPGTUD, do_vcmp, TCG_COND_GTU, MO_64)
+
+TRANS(VCMPNEB, do_vcmp, TCG_COND_NE, MO_8)
+TRANS(VCMPNEH, do_vcmp, TCG_COND_NE, MO_16)
+TRANS(VCMPNEW, do_vcmp, TCG_COND_NE, MO_32)
+
+static void gen_vcmpnez_vec(unsigned vece, TCGv_vec t, TCGv_vec a, TCGv_vec b)
+{
+    TCGv_vec t0, t1, zero;
+
+    t0 = tcg_temp_new_vec_matching(t);
+    t1 = tcg_temp_new_vec_matching(t);
+    zero = tcg_constant_vec_matching(t, vece, 0);
+
+    tcg_gen_cmp_vec(TCG_COND_EQ, vece, t0, a, zero);
+    tcg_gen_cmp_vec(TCG_COND_EQ, vece, t1, b, zero);
+    tcg_gen_cmp_vec(TCG_COND_NE, vece, t, a, b);
+
+    tcg_gen_or_vec(vece, t, t, t0);
+    tcg_gen_or_vec(vece, t, t, t1);
+
+    tcg_gen_shli_vec(vece, t, t, (8 << vece) - 1);
+    tcg_gen_sari_vec(vece, t, t, (8 << vece) - 1);
+
+    tcg_temp_free_vec(t0);
+    tcg_temp_free_vec(t1);
+}
+
+static bool do_vcmpnez(DisasContext *ctx, arg_VC *a, int vece)
+{
+    static const TCGOpcode vecop_list[] = {
+        INDEX_op_cmp_vec, INDEX_op_shli_vec, INDEX_op_sari_vec, 0
+    };
+    static const GVecGen3 ops[3] = {
+        {
+            .fniv = gen_vcmpnez_vec,
+            .fno = gen_helper_VCMPNEZB,
+            .opt_opc = vecop_list,
+            .vece = MO_8
+        },
+        {
+            .fniv = gen_vcmpnez_vec,
+            .fno = gen_helper_VCMPNEZH,
+            .opt_opc = vecop_list,
+            .vece = MO_16
+        },
+        {
+            .fniv = gen_vcmpnez_vec,
+            .fno = gen_helper_VCMPNEZW,
+            .opt_opc = vecop_list,
+            .vece = MO_32
+        }
+    };
+
+    tcg_gen_gvec_3(avr_full_offset(a->vrt), avr_full_offset(a->vra),
+                   avr_full_offset(a->vrb), 16, 16, &ops[vece]);
+
+    if (a->rc) {
+        do_vcmp_rc(a->vrt);
+    }
+
+    return true;
+}
+
+TRANS(VCMPNEZB, do_vcmpnez, MO_8)
+TRANS(VCMPNEZH, do_vcmpnez, MO_16)
+TRANS(VCMPNEZW, do_vcmpnez, MO_32)
+
+static bool trans_VCMPEQUQ(DisasContext *ctx, arg_VC *a)
+{
+    TCGv_i64 t0, t1;
+    TCGLabel *l1, *l2;
+
```

Comment:
Isn't it necessary to check for `REQUIRE_INSNS_FLAGS2(ctx, ISA310);` and `REQUIRE_VECTOR(ctx);` ?

Reply:
Fixed in b005ed5712d130f3ed5db2743be273d9a510bbca